### PR TITLE
document flytekit remote module

### PR DESCRIPTION
**Pull request by cosmicBboy**
Signed-off-by: cosmicBboy <niels.bantilan@gmail.com>

# TL;DR

- Add documentation for `remote` module
- change `FlyteRemote.from_environment` to `from_config`
- add `@property` vars for FlyteRemote properties
- improve doc strings
- add a little more content to flytekit home page

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin
 - [X] Docs

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
NA

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: docs/source/design/control_plane.rst [hunk 1]
```diff
@@ -1,55 +1,95 @@
...
+********************************
 
-Workflow
-========
-To inspect a Workflow ::
+.. code-block:: python
 
-    from flytekit.common.workflow import SdkWorkflow
-    wf = SdkWorkflow.fetch('flytetester', 'staging', 'recipes.core_basic.basic_workflow.my_wf', '49b6c6bdbb86e974ffd9875cab1f721ada8066a7')
+    completed_execution = remote.wait(execution)
```
Comment (katrogan) [inline docs/source/design/control_plane.rst:69]: is this synchronous too? worth calling out?

File: flytekit/remote/remote.py [hunk 2]
```diff
@@ -566,14 +620,13 @@ def execute(
...
-        :param execution_name: name of the execution
+        :param execution_name: name of the execution. If None, uses auto-generated value.
         :param wait: if True, waits for execution to complete
-        :returns: :class:`~flytekit.remote.workflow_execution.FlyteWorkflowExecution`
 
         .. note::
 
-            The ``project``, ``domain``, ``name``. and ``version`` arguments do not apply to ``FlyteTask``,
-            ``FlyteLaunchPlan``, and ``FlyteWorkflow`` objects.
+            The ``project``, ``domain``, ``name`` and ``version`` arguments do not apply to ``FlyteTask``,
```
Comment (katrogan) [inline flytekit/remote/remote.py:628]: does this mean users can never override the project or domain that an execution is launched under?

**Review by katrogan — APPROVED** [review]

File: docs/source/design/control_plane.rst [hunk 3]
```diff
@@ -4,18 +4,17 @@
...
-backend set up, and have access to it (a local sandbox will suffice as well of course). These objects do not rely on the
-underlying code they represent being locally available.
+backend set up and have access to it (a :std:ref:`local sandbox <flyte:deployment-sandbox>` will suffice as well).
 
 ***************************
 Create a FlyteRemote Object
 ***************************
 
-The :class:`~flytekit.remote.FlyteRemote` class is the entrypoint for programmatically performing operations in a python
+The :class:`~flytekit.remote.remote.FlyteRemote` class is the entrypoint for programmatically performing operations in a python
```
Comment (katrogan) [inline docs/source/design/control_plane.rst:17]: nit: why the double remote in the path? isn't it exposed in flytekit/remote/__init__.py ?

File: flytekit/remote/remote.py [hunk 4]
```diff
@@ -648,10 +680,17 @@ def _(
     ) -> FlyteWorkflowExecution:
         """Execute a FlyteTask, or FlyteLaunchplan.
 
-        NOTE: the project, domain, name, and version arguments are currently not used and only there consistency in
-        the function signature
+        NOTE: the name and version arguments are currently not used and only there consistency in the function signature
```
Comment (katrogan) [inline flytekit/remote/remote.py:683]: should we maybe log or raise an error is they are accidentally supplied?

**Review by katrogan — APPROVED** [review]
lgtm, just some nits - sorry!

File: docs/source/design/control_plane.rst (hunk 3 above)
Comment (cosmicBboy) [reply docs/source/design/control_plane.rst:17]: yes, but sphinx needs the fully qualified path for the link to show up

**Review by katrogan — APPROVED** [review]

**Comment by cosmicBboy** [comment]
> oh shoot, sorry for being annoying, i didn't realize how many places flyte_id_kwargs were being used as a dict

no worries!

**Pull request merged**